### PR TITLE
Refactored RouteOptions integration

### PR DESCRIPTION
**Pull request by korshaknn**
## Description

Draft PR with changes needed to integrate refactored `RouteOptions` and `MapboxDirections`
- [ ] I have added any issue links
- [ ] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal

Please describe the PR goals. Just the stuff needed to implement the fix / feature and a simple rationale. It could contain many check points if needed

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/build.gradle [hunk 1]
```diff
@@ -57,6 +57,7 @@ android {
 
 dependencies {
   api dependenciesList.mapboxSdkServices
+  api dependenciesList.mapboxSdkDirectionsModels
```
Comment (LukasPaczos) [inline libandroid-navigation/build.gradle]: Do we need this dependency? It should come together with `mapboxSdkServices`.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 2]
```diff
@@ -424,27 +423,15 @@ class MapboxNavigation(
...
-                        Pair(
-                            location.bearing.toDouble(),
-                            originTolerance ?: DEFAULT_REROUTE_BEARING_TOLERANCE
-                        )
-                    )
-                    optionsBuilder.bearings(*pairs.toTypedArray())
-                }
+                val bearings = mutableListOf<List<Double>>()
+
+                val originTolerance = options.bearingsList()?.get(0)?.get(1) ?: DEFAULT_REROUTE_BEARING_TOLERANCE
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: the bearing might not have been passed with the original request, so we need to keep using `getOrNull`

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 3]
```diff
@@ -424,27 +423,15 @@ class MapboxNavigation(
...
+                val originAngle = location.bearing.toDouble()
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: NIT this is the current angle, not original angle

File: libandroid-navigation/build.gradle (hunk 1 above)
Comment (Guardiola31337) [reply libandroid-navigation/build.gradle]: Yeah, agreed. Did you add this for testing purposes (explicitly bring the `SNAPSHOT` version in `mapboxSdkDirectionsModels`) @korshaknn? If so, we should remove before merging the PR 👍

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 2 above)
Comment (korshaknn) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: fixed

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 3 above)
Comment (korshaknn) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt]: fixed

File: gradle/dependencies.gradle [hunk 4]
```diff
@@ -10,7 +10,8 @@ ext {
   version = [
       mapboxMapSdk              : '9.1.0-SNAPSHOT',
       mapboxSdkServices         : '5.1.0-beta.1',
-      mapboxSdkDirectionsModels : '5.1.0-beta.1',
+      mapboxSdkServicesSnapshot : '5.1.0-20200217.082443-24',
+      mapboxSdkDirectionsModels : '5.1.0-20200217.082455-24',
```
Comment (korshaknn) [inline gradle/dependencies.gradle]: @LukasPaczos  @Guardiola31337  
is it ok, that `mapboxSdkServices` use a new  dependency `mapboxSdkServicesSnapshot : '5.1.0-20200217.082443-24'`

but `mapboxSdkGeoJSON ` and `mapboxSdkTurf` still use 
mapboxSdkServices         : '5.1.0-beta.1' . ?

or they all should use a new snapshot dependency?

File: libandroid-navigation/build.gradle (hunk 1 above)
Comment (korshaknn) [reply libandroid-navigation/build.gradle]: fixed

File: gradle/dependencies.gradle (hunk 4 above)
Comment (Guardiola31337) [reply gradle/dependencies.gradle]: Ideally we should release a new beta of MAS including the latest changes. Could you run with that @korshaknn? cc @langsmith

Comment (korshaknn) [reply gradle/dependencies.gradle]: @Guardiola31337 haven't released it before, don't know the flow

Comment (LukasPaczos) [reply gradle/dependencies.gradle]: For now, we could also add `5.1.0-SNAPSHOT` to both `mapboxSdkDirectionsModels` and `mapboxSdkServices` versions.

Comment (korshaknn) [reply gradle/dependencies.gradle]: @LukasPaczos fixed

**Comment by korshaknn** [comment]
@LukasPaczos  @Guardiola31337 ptal

**Pull request merged**